Sources/PaylikeEngine/ViewModel/Engine/PaylikeEngineAddPaymentData.swift

```diff
@@ -0,0 +1,144 @@
+            let card = try await PaymentCard(number: numberToken, code: cvcToken, expiry: expiry)
+            paymentRepository.card = card
+                    
+            await savePaymentRepository(newRepository: paymentRepository)
+            
+            loggingFn(Loggingformat(t: "Added CardData essential payment data", state: self.state, paymentData: paymentRepository))
+        } catch {
+            prepareError(e: error)
+        }
+    }
+    
+    /**
+     * These fields describe the payment characteristics. To set up check the api docs below.
+     * @param paymentAmount define a single payment amount
+     * @param paymentPlanDataList define reoccurring payments
+     * @param paymentUnplannedData define the types of unplanned payments the card will be used for
+     * [More information](https://github.com/paylike/api-reference/blob/main/payments/index.md)
+     */
+    public func addDescriptionPaymentData(
+        paymentAmount: PaymentAmount? = nil,
+        paymentPlanDataList: [PaymentPlan]? = nil,
+        paymentUnplannedData: PaymentUnplanned? = nil,
+        paymentTestData: PaymentTest? = nil
+    ) {
+        do {
+            try checkValidState(valid: EngineState.WAITING_FOR_INPUT, callerFunc: #function)
+            var paymentRepository = initialisePaymentRepositoryIfNil()
+            
+            paymentRepository.amount = paymentAmount
+            paymentRepository.plan = paymentPlanDataList
+            paymentRepository.unplanned = paymentUnplannedData
+            paymentRepository.test = paymentTestData
+            
+            savePaymentRepository(newRepository: paymentRepository)
+            
+            loggingFn(Loggingformat(t: "Added description payment data", state: self.state, paymentData: paymentRepository))
+        } catch {
+            prepareError(e: error)
+        }
+    }
+    
+    /**
+     * These field are optional to define.
+     * @param textData is a simple text shown on the paylike dashboard
+     * @param customData is a custom Json object defined by the user
+     * [More information](https://github.com/paylike/api-reference/blob/main/payments/index.md)
+     */
+    public func addAdditionalPaymentData(
+        textData: String? = nil,
+        customData: AnyEncodable? = nil
+    ) {
+        do {
+            try checkValidState(valid: EngineState.WAITING_FOR_INPUT, callerFunc: #function)
+            var paymentRepository = initialisePaymentRepositoryIfNil()
+            
+            paymentRepository.text = textData
+            paymentRepository.custom = customData
+            
+            savePaymentRepository(newRepository: paymentRepository)
+            
+            loggingFn(Loggingformat(t: "Added additional payment data", state: self.state, paymentData: paymentRepository))
+        } catch {
+            prepareError(e: error)
+        }
+    }
+    
+    private func initialisePaymentRepositoryIfNil() -> CreatePaymentRequest {
```

Comment:
Should this be here? In terms of logical context, should not this be in `PaylikeEngine.swift` ?

Reply:
This function is only called in this file. So it could have `filePrivate` accessibility.

So I left it in this file because this is the "scope" where it's used, where it's a helper function.

It feels a bit separated from the original scope moving it to the main file. But I see the point of your question.

What do you think about changing it to `filePrivate`?